### PR TITLE
add Budget class, budget and showbudget command

### PR DESCRIPTION
**Pull request by austinsantoso**
add budget class, with save and load functionalities. added budget and showbudget commands

### CODE REVIEW

File: src/test/java/seedu/address/logic/commands/BudgetCommandTest.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+import seedu.address.model.ModelManager;
+import seedu.address.model.UserPrefs;
+import seedu.address.model.budget.Budget;
+
+class BudgetCommandTest {
+
+    private Model model = new ModelManager(getTypicalAddressBook(), new UserPrefs());
+
+    @Test
+    void execute_setNewBudget_success() {
```
Comment (choongyx) [inline src/test/java/seedu/address/logic/commands/BudgetCommandTest.java]: public void --> to remain consistent with other tests and for code quality check

File: src/test/java/seedu/address/logic/commands/ShowBudgetCommandTest.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+import seedu.address.model.Model;
+import seedu.address.model.ModelManager;
+import seedu.address.model.UserPrefs;
+
+class ShowBudgetCommandTest {
+
+    private Model model = new ModelManager(getTypicalAddressBook(), new UserPrefs());
+
+    @Test
+    void execute_displayBudget_success() {
```
Comment (choongyx) [inline src/test/java/seedu/address/logic/commands/ShowBudgetCommandTest.java]: public void --> to remain consistent with other tests and for code quality check

File: src/main/java/seedu/address/model/budget/Budget.java [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+/**
+ * Represents the budget in the address book.
+ */
+public class Budget {
+
+    public static final String MESSAGE_CONSTRAINTS = "Budget values should be positive number";
```
Comment (choongyx) [inline src/main/java/seedu/address/model/budget/Budget.java:11]: Are we allowing 2dp for budget (like Cost) or budget can use Cost too?

**Comment by choongyx** [comment]
#30

File: src/main/java/seedu/address/logic/commands/ShowBudgetCommand.java [hunk 4]
```diff
@@ -0,0 +1,18 @@
+package seedu.address.logic.commands;
+
+import seedu.address.model.Model;
+
+/**
+ * Displays the current SPending
```
Comment (Nanosync) [inline src/main/java/seedu/address/logic/commands/ShowBudgetCommand.java]: Budget

File: src/test/java/seedu/address/logic/parser/BudgetCommandParserTest.java [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+
+    @Test
+    public void parse_validArgs_returnBudgetCommand() {
+        Budget validBudget = new Budget(12);
+        assertParseSuccess(parser, "12", new BudgetCommand(validBudget));
+    }
+
+    @Test
+    public void parse_invalidArgs_throwsParseException() {
+        assertParseFailure(parser, "", String.format(MESSAGE_INVALID_COMMAND_FORMAT,
```
Comment (Nanosync) [inline src/test/java/seedu/address/logic/parser/BudgetCommandParserTest.java:32]: add a negative value argument for test case

**Review by choongyx — APPROVED** [review]

**Review by Nanosync — APPROVED** [review]

**Pull request merged**